config/server.py

```diff
@@ -0,0 +1,196 @@
+import uuid
+import argparse
+from urllib.parse import urlparse
+from functools import wraps
+
+import connexion
+import flask
+from google.oauth2 import id_token
+from google.auth.transport import requests
+from peewee import CharField, Model, SqliteDatabase
+
+
+def get_route_model(db):
+    class Route(Model):
+        owner = CharField()
+        name = CharField()
+        destination = CharField()
+        token = CharField()
+
+        def get_json(self):
+            return {
+                "owner": self.owner,
+                "destination": self.destination,
+                "token": self.token,
+                "name": self.name
+            }
+
+        class Meta:
+            database = db
+
+    return Route
+
+
+class InvalidCredentials(Exception):
+    pass
+
+
+class NotAuthorised(Exception):
+    pass
+
+
+class InvalidRouteID(Exception):
+    pass
+
+
+class InvalidURL(Exception):
+    pass
+
+google_oauth_clientID = "859663336690-q39h2o7j9o2d2vdeq1hm1815uqjfj5c9.self.apps.googleusercontent.com"
+
+class Auth:
+    def get_user(self):
+        token = flask.request.headers.get("Google-Auth-Token")
+
+        if token is None:
+            raise InvalidCredentials()
+
+        try:
+            token_info = id_token.verify_oauth2_token(token, requests.Request(), google_oauth_clientID)
+        except ValueError as ve:
+            raise InvalidCredentials() from ve
+
+        if token_info["hd"] != "sanger.ac.uk":
+            raise InvalidCredentials()
+
+        if token_info['iss'] not in ['accounts.google.com', 'https://accounts.google.com']:
+            raise InvalidCredentials()
+
+        return token_info["email"]
+
+class Server:
+    def _auth_request(func):
+        @wraps(func)
+        def new_func(self, *args, **kw_args):
+            current_user = self.auth.get_user()  # includes auth
+
+            self._log_function_call(func.__name__, current_user, args)
+
+            return func(self, *args, **kw_args)
+
+        return new_func
+
+    def _log_function_call(self, name, user, parameters):
+        # TODO
+        pass
+
+    def _token2route(self, token: str) -> get_route_model(None):
```

Comment:
I read this as "token-to-route"; that is a convertor of some description.

Reply:
2 != to

Reply:
Be that as it may, it's a *really* common naming scheme for utilities. For example, on my machine, there are (heuristically) 40 such commands:

```bash
$ compgen -c | grep -E "^[a-z]{3,4}2[a-z]{3,4}$" | sort | uniq | wc -l
      40
```

...from `ace2sam` to `xml2man` 🙂

Reply:
(I do agree that it should have a more descriptive name, though!)